fxtest/fxtest.go

```diff
@@ -125,10 +130,12 @@ func (l *Lifecycle) RequireStart() *Lifecycle {
 // returned.
 func (l *Lifecycle) Stop(ctx context.Context) error { return l.lc.Stop(ctx) }
 
-// RequireStop calls Stop with context.Background(), failing the test if an error
+// RequireStop calls Stop and cancels the context passed in to Start, failing the test if an error
 // is encountered.
 func (l *Lifecycle) RequireStop() {
-	if err := l.Stop(context.Background()); err != nil {
+	defer l.cancelFunc()
+
+	if err := l.Stop(l.ctx); err != nil {
```

Comment:
Sharing the context between `RequireStart` and `RequireStop` will end up behaving differently than the production implementation. In particular, constructors shouldn't rely on the context passed to `Start` being cancelled during `Stop` - they should be completely independent.

This also simplifies this diff - we can just replace 

```
if err := l.Start(context.Background()); err != nil {
```
in `RequireStart` and `RequireStop` with 
```
ctx, cancel := context.WithCancel(context.Background)
defer cancel()

if err := l.Start(ctx); err != nil {
```

We shouldn't need to add fields to the `Lifecycle` structure at all.

Reply:
Perfect, I was wondering if this worked out like this - but you writing this confirms my previous thought. I have updated to be a self contained cancelable context.